### PR TITLE
Frontend refactoring

### PR DESCRIPTION
**Pull request by tameware**
It would be fine to review this pull instead of 'frontend'. This one is a strict refactoring of that one - there are no functionality changes.

### CODE REVIEW

File: frontend/dds_mvp.js [hunk 1]
```diff
@@ -54,90 +69,124 @@ function fillFormWithPartScoreTestData() {
...
 
+function sendJSON() {
     var hands = collectHands();
-    var deal = { "hands": hands };
-    var data = JSON.stringify(deal);
-    xhr.send(data);
+
+    const error_message = inputIsValid(hands);
+
+    if (error_message.length == 0) {
```
Comment (suokko) [inline frontend/dds_mvp.js:166]: A good practice would be having error check only having failure branch which returns. Then rest of function body would avoid increased indention. In this case something like:

if (error_messa.length != 0) {
document.getElementById("result").innerHTML = error_message;
return;
}

File: frontend/dds_mvp.js [hunk 2]
```diff
@@ -54,90 +69,124 @@ function fillFormWithPartScoreTestData() {
...
 
-    xhr.setRequestHeader("Content-Type", "application/json");
+        if (hand.length != 13) {
+            return "Please enter 13 cards per hand.";
+        }
 
-    xhr.onreadystatechange = function () {
-        if (xhr.readyState === 4 && xhr.status === 200) {
-            document.getElementById("result").innerHTML = this.responseText;
+        for (const card of hand) {
```
[the remark this answers is not in the document]
Comment (tameware) [reply frontend/dds_mvp.js:146]: Agreed. That will be in my next pull request. I think I left a TODO for it.

File: frontend/dds_mvp.js [hunk 3]
```diff
@@ -54,90 +69,124 @@ function fillFormWithPartScoreTestData() {
...
 }
 
-function sendJSON(){
-    var xhr = new XMLHttpRequest();
-    const URL = "http://localhost:5000/api/dds-table/";
-
-    // This fails as of 2020-06-21 due to:
-    // Access from origin 'null' has been blocked by CORS policy
-    // const URL = "https://dds.hackathon.globalbridge.app/api/dds-table/";
+function inputIsValid(hands) {
```
[the remark this answers is not in the document]
Comment (tameware) [reply frontend/dds_mvp.js:136]: Yes, we have an open issue for that.

File: frontend/dds_mvp.js [hunk 4]
```diff
@@ -54,90 +69,124 @@ function fillFormWithPartScoreTestData() {
...
+    if (error_message.length == 0) {
+        var xhr = new XMLHttpRequest();
+        const URL = "http://localhost:5000/api/dds-table/";
+
+        // This fails as of 2020-06-21 due to:
+        // Access from origin 'null' has been blocked by CORS policy
+        // That's because an older version of our service is deployed.
+        // const URL = "https://dds.hackathon.globalbridge.app/api/dds-table/";
+
+        xhr.open("POST", URL, true);
```
[the remark this answers is not in the document]
Comment (tameware) [reply frontend/dds_mvp.js:175]: We have no such plans at present, but I agree that it's a good idea. Please suggest it on our Discord channel!

File: frontend/dds_mvp.js (hunk 1 above)
Comment (tameware) [reply frontend/dds_mvp.js:166]: Will do.

Comment (tameware) [reply frontend/dds_mvp.js]: Done and pushed. Please have another look.

**Pull request merged**